picodom.d.ts

```diff
@@ -1,26 +1,41 @@
-export interface VNode<Data> {
+export as namespace picodom
+
+export interface VirtualNode<Data> {
   tag: string
-  data?: Data
+  data: Data
   children: VElement<{}>[]
 }
 
 export interface Component<Data> {
-  (data?: Data, ...children: VElement<{}>[]): VNode<Data>
+  (data: Data, children: VElement<{}>[]): VirtualNode<any>
 }
 
-export type VElement<Data> = VNode<Data> | string
+export type VElement<Data> = VirtualNode<Data> | string
+
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  children?: Array<VElement<{}> | number>
+): VirtualNode<Data>
 
 export function h<Data>(
   tag: Component<Data> | string,
   data?: Data,
-  ...children: VElement<{}>[]
-): VNode<Data>
+  ...children: Array<VElement<{}> | number>
```

Comment:
Let's keep the discussion of whether or not to keep the argument spread version of h in [hyperapp PR 311](https://github.com/hyperapp/hyperapp/pull/311) where it is currently happening. I left it here since it already existed.